src/petals/server/handler.py

```diff
@@ -142,6 +144,7 @@ async def rpc_inference(
                 requested_backends = tuple(self.module_backends[uid] for uid in requested_uids)
                 max_length = metadata.get("max_length")
                 active_adapter = metadata.get("active_adapter", "")
+                assert not active_adapter or active_adapter in self.adapters
```

Comment:
Please add a error message here, otherwise the client will receive P2PHandlerError("") and won't know what happened